### PR TITLE
Enable CA1068

### PR DESCRIPTION
**Pull request by davidCett**
# Pull Request

## The issue or feature being addressed

Cleanup the Polly and Polly.Specs codebase #1290

## Details on the issue fix or feature implementation

Enable CA1068 warning (CancellationToken parameters must come last) and code cleanup

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

**Comment by davidCett** [comment]
@dotnet-policy-service agree

File: src/Polly/AsyncPolicy.GenericImplementation.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public abstract partial class AsyncPolicy<TResult>
     protected abstract Task<TResult> ImplementationAsync(
         Func<Context, CancellationToken, Task<TResult>> action,
         Context context,
-        CancellationToken cancellationToken,
-        bool continueOnCapturedContext
+        bool continueOnCapturedContext,
```
Comment (martincostello) [inline src/Polly/AsyncPolicy.GenericImplementation.cs:16]: Just scan reading this PR before a proper review tomorrow, the public API surface area like this **must not** be changed to resolve these warnings. Anything this isn't internal or private must be suppressed only.

Changing the order of the parameters in the public API surface is a breaking change we cannot accept.

**Review by martincostello — CHANGES_REQUESTED** [review]

**Comment by martincostello** [comment]
As you can [see in the build log](https://github.com/App-vNext/Polly/actions/runs/6409461852/job/17403157798?pr=1677#step:5:674), the Public API analyser flags the changes to the public surface of the library and breaks the build.

For example:

```
Error: C:\Program Files\dotnet\sdk\7.0.401\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0002: Member 'Polly.AsyncPolicy.ExecuteAsync(System.Func<System.Threading.CancellationToken, System.Threading.Tasks.Task>, System.Threading.CancellationToken, bool)' exists on [Baseline] lib/net461/Polly.dll but not on lib/netstandard2.0/Polly.dll [D:\a\Polly\Polly\src\Polly\Polly.csproj]
```

File: src/Polly/PublicAPI.Shipped.txt
Comment (martincostello) [inline src/Polly/PublicAPI.Shipped.txt:1]: This is the wrong thing to do - the file is effectively set in stone and should not be edited.

Comment (davidCett) [reply src/Polly/PublicAPI.Shipped.txt:1]: I get it. My bad.

**Comment by davidCett** [comment]
Thank you for de review. I will work on it😀

**Pull request closed without merging**